@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,194 @@ export namespace MapViewUtils {
+            ninetyRad * 2 -
+            (ninetyRad + angleBetweenNormals - rotationCenterTilt + ninetyRad + tilt);
+
+        // This uses the same formula as above, just we change `tilt` to the maxTiltAngle,
+        // i.e we want to find the greatest angle at the rotation target that gives us the max
+        // angle at the map center target.
+        // i.e. CMR is 90 + `maxTiltAngle`
+        const maxRotationTargetAngle =
+            ninetyRad * 2 - angleBetweenRotationTargetAndMapTarget - ninetyRad - maxTiltAngle;
+
+        // Converting the `maxRotationTargetAngle` back to a tilt is as easy as subtracting it from
+        // 90 and the `angleBetweenNormals`, i.e. this gives us the maximum allowed tilt at R that
+        // satisfies the `maxTiltAngle` constraint. Note, for globe projection, this is just an
+        // approximation, because once we move the camera by delta, the map target changes, and
+        // therefore the normal also changes, this would need to be applied iteratively until the
+        // difference in normals is reduced to some epsilon. I don't apply this because it is
+        // computationally expensive and the user would never notice this in practice.
+        const maxTilt = ninetyRad + angleBetweenNormals - maxRotationTargetAngle;
+
+        // Here we clamp to the min of `maxTilt` and 89 degrees. The check for 89 is to prevent it
+        // intersecting with the world at 90. This is possible for example when the R position is
+        // near the horizon. If the angle RCM is say 5 degrees, then an angle of say 89 degrees at
+        // R, plus 5 degrees means the tilt at M would be 84 degrees. If the `maxTiltAngle` is
+        // however 89, then we still have 5 degrees at M, so, without this check, we would go to 90
+        // degrees.
+        const clampedDeltaTilt =
+            MathUtils.clamp(
+                deltaTilt + rotationCenterTilt,
+                0,
+                Math.min(maxTilt, maxRotationTiltAngle)
+            ) - rotationCenterTilt;
+
+        return clampedDeltaTilt;
+    }
+
+    /**
+     * @hidden
+     * @internal
+     *
+     * Applies the given tilt to the camera around the supplied target.
+     */
+    function applyTiltAroundTarget(
+        mapView: MapView,
+        rotationTargetWorld: THREE.Vector3,
+        deltaTilt: number,
+        tiltAxis: THREE.Vector3
+    ) {
+        const camera = mapView.camera;
+        // Consider to use the cache if necessary, but beware, because the `rayCastWorldCoordinates`
+        // also uses this cache.
+        const posBackup = camera.position.clone();
+        const quatBackup = camera.quaternion.clone();
 
-        const tiltAxis = cache.vector3[0].set(1, 0, 0).applyQuaternion(camera.quaternion);
-        const tiltQuat = cache.quaternions[0].setFromAxisAngle(tiltAxis, clampedDeltaTilt);
+        const tiltQuat = cache.quaternions[0].setFromAxisAngle(tiltAxis, deltaTilt);
```

Comment:
Just a comment in case we come back to this change in the future. I thought it makes more sense to use as tilt axis the cross product of the up camera vector with the direction vector from the camera to the rotation center.
I've tried it and it works (it keeps the rotation center in place), the only problem is that it rolls the camera. Maybe we could investigate how to undo or avoid this roll.

Reply:
Please make a ticket.

Reply:
so far we don't need to do anything, as long as this works right. Not sure we need a ticket, it was just a comment for reference.

Reply:
I made a ticket: HARP-12600